pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -493,6 +493,10 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 						}
 
 						shoot.Status.EncryptedResources = encryptedResources
+						if shoot.Status.Credentials == nil {
```

Comment:
Can we avoid creating the empty `Credentials` field if `encryptedResources` is empty?